File: sohy/190728_detect_capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        
        char[] charWord = word.toCharArray();
        int numOfCapital = 0; 
        for (char eachWord: charWord) {
            if (Character.isUpperCase(eachWord)) numOfCapital ++; 
        }
        
        if (numOfCapital == 0) return true;
        else if (numOfCapital == 1 && Character.isUpperCase(charWord[0])) return true; 
        else if (numOfCapital == word.length()) return true; 
        return false; 
    }
}

/*
Runtime: 1 ms, faster than 100.00% of Java online submissions for Detect Capital.
Memory Usage: 34.8 MB, less than 100.00% of Java online submissions for Detect Capital.

*/

File: 2kindsofcs/190728-detect-capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if word.upper() == word:
            return True
        if word.lower() == word:
            return True
        if word[0].upper() == word[0] and word[1:].lower() == word[1:]:
            return True
        return False
    
# Runtime: 24 ms, faster than 100.00% of Python3 online submissions for Detect Capital.
# Memory Usage: 13.7 MB, less than 5.44% of Python3 online submissions for Detect Capital.
